src/app/doi.org/[...doi]/RelatedAggregateGraph.tsx

```diff
@@ -10,14 +10,45 @@ interface Props {
   isBot?: boolean
 }
 
+interface GraphData {
+  nodes: any[];
+  links: any[];
+}
+
+interface FetchResult {
+  data: GraphData;
+  timedOut: boolean;
+}
+
+async function fetchRelatedWorksGraphWithTimeout(doi: string, timeoutDuration: number): Promise<FetchResult> {
+  function timeoutPromise(ms: number) {
+    return new Promise<never>((_, reject) => setTimeout(() => reject(new Error('timeout')), ms));
+  }
+
+  try {
+    const data = await Promise.race([
+      getRelatedWorksGraph(doi),
+      timeoutPromise(timeoutDuration)
+    ]);
+    return { data, timedOut: false };
+  } catch (error) {
+    console.error("Request timed out");
```

Comment:
Do we want to keep this console.error?

Reply:
Good catch.  It is on the server side and would get printed to the vercel logs.

In general I think it is a good practice to remove any console errors.
Expect an update